### DISCUSSION TITLE
setup.py error .. how to solve it pls

### DISCUSSION THREAD

**Discussion by MAHDIATH89** [datasciencecompleteguide/Machine-Learning#50] [Q&A]
C:\Users\Admin>pip install dlib
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15)")': /simple/dlib/
Collecting dlib
  Using cached dlib-19.24.0.tar.gz (3.2 MB)
  Preparing metadata (setup.py) ... done
Using legacy 'setup.py install' for dlib, since package 'wheel' is not installed.
Installing collected packages: dlib
  Running setup.py install for dlib ... error
  error: subprocess-exited-with-error

  × Running setup.py install for dlib did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      running install
      running build
      running build_py
      package init file 'tools\python\dlib\__init__.py' not found (or not a regular file)
      running build_ext

      ERROR: CMake must be installed to build dlib

      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> dlib

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

C:\Users\Admin>

**Comment by datasciencecompleteguide** [comment]
Have you installed CMAKE properly as mentioned in video....because I can see in error
 ```
 ERROR: CMake must be installed to build dlib

  [end of output]
```
Thus, check this problem first. Even if you have installed the CMAKE, then also uninstall it and delete all the files than reinstall it...because here is the dependency error.

Next, Step I would suggest to you is that use the lower version of python (I will prefer 3.8.6).
